src/pybind/mgr/rook/rook_cluster.py

```diff
@@ -243,20 +259,73 @@ def rook_api_post(self, path: str, **kwargs: Any) -> Any:
     def get_discovered_devices(self, nodenames: Optional[List[str]] = None) -> Dict[str, dict]:
         def predicate(item: client.V1ConfigMapList) -> bool:
             if nodenames is not None:
-                return item.metadata.labels['rook.io/node'] in nodenames
+                return item['spec']['nodeName'] in nodenames
             else:
                 return True
+        matching_sc = [i for i in self.storage_classes.items if self.storage_class == i.metadata.name]
+        if len(matching_sc) == 0:
+            log.exception("no storage class exists matching name provided in ceph config at mgr/rook/storage_class")
+            raise Exception('No storage class exists matching name provided in ceph config at mgr/rook/storage_class')
+        if len(matching_sc) > 1:
+            log.exception("too many storage classes matching name provided in ceph config at mgr/rook/storage_class")
```

Comment:
the "exception" method for log object provide a full stack trace of the error. In this case, the error is a "logical" error .. the stack trace is not useful at all.
I think better to use "log.error"